android-components/components/browser/state/src/test/java/mozilla/components/browser/state/state/TranslationEngineStateTest.kt

```diff
@@ -0,0 +1,121 @@
+            userPreferredLangTag = "es",
+        )
+
+        val translationEngineState = TranslationEngineState(
+            detectedLanguages = detectedLanguages,
+            error = null,
+            isEngineReady = true,
+            requestedTranslationPair = null,
+        )
+
+        assert(translationEngineState.initialToLanguage(mockCandidateLanguages)?.code == "es")
+        assert(translationEngineState.initialFromLanguage(mockCandidateLanguages)?.code == "en")
+    }
+
+    @Test
+    fun `initialToLanguage and initialFromLanguage works as expected on the translated case`() {
+        val detectedLanguages = DetectedLanguages(
+            documentLangTag = "en",
+            supportedDocumentLang = true,
+            userPreferredLangTag = "es",
+        )
+
+        val translationPair = TranslationPair(
+            fromLanguage = "fr",
+            toLanguage = "de",
+        )
+
+        val translationEngineState = TranslationEngineState(
+            detectedLanguages = detectedLanguages,
+            error = null,
+            isEngineReady = true,
+            requestedTranslationPair = translationPair,
+        )
+
+        assert(translationEngineState.initialToLanguage(mockCandidateLanguages)?.code == "de")
+        assert(translationEngineState.initialFromLanguage(mockCandidateLanguages)?.code == "fr")
+    }
+
+    @Test
+    fun `initialToLanguage and initialFromLanguage works as expected on the expected null case when not translated`() {
+        val detectedLanguages = DetectedLanguages(
+            documentLangTag = "not-a-code",
+            supportedDocumentLang = true,
+            userPreferredLangTag = "not-a-code",
+        )
+
+        val translationEngineState = TranslationEngineState(
+            detectedLanguages = detectedLanguages,
+            error = null,
+            isEngineReady = true,
+            requestedTranslationPair = null,
+        )
+
+        assert(translationEngineState.initialToLanguage(mockCandidateLanguages) == null)
+        assert(translationEngineState.initialFromLanguage(mockCandidateLanguages) == null)
+    }
+
+    @Test
+    fun `initialToLanguage and initialFromLanguage works as expected on the expected null case when translated`() {
+        val detectedLanguages = DetectedLanguages(
+            documentLangTag = "en",
+            supportedDocumentLang = true,
+            userPreferredLangTag = "es",
+        )
+
+        // This would be a highly unexpected state
+        val translationPair = TranslationPair(
+            fromLanguage = "not-a-code",
+            toLanguage = "not-a-code",
+        )
+
+        val translationEngineState = TranslationEngineState(
+            detectedLanguages = detectedLanguages,
+            error = null,
+            isEngineReady = true,
+            requestedTranslationPair = translationPair,
+        )
+
+        assert(translationEngineState.initialToLanguage(mockCandidateLanguages) == null)
```

Comment:
```suggestion
        assert(translationEngineState.initialToLanguage(mockCandidateLanguages) == null)
```
You could also simply this to use `assertNull`